ovn_cluster.sh

```diff
@@ -484,13 +587,15 @@ function start() {
 
     # Create containers
     if [ "$ovn_central" == "yes" ]; then
-        if [ "$OVN_DB_CLUSTER" = "yes" ]; then
-            start-container "${CENTRAL_IMAGE}" "${CENTRAL_NAME}-1"
-            start-container "${CENTRAL_IMAGE}" "${CENTRAL_NAME}-2"
-            start-container "${CENTRAL_IMAGE}" "${CENTRAL_NAME}-3"
-        else
-            start-container "${CENTRAL_IMAGE}" "${CENTRAL_NAME}"
-        fi
+        for name in "${CENTRAL_NAMES[@]}"; do
```

Comment:
We shouldn't check for central containers not running (`check-no-containers()`) if ovn_central is "no".

I think what we need to do is split check-no-containers() into different check functions and call them above, e.g.:

```
function check-no-central {
  local operation=$1
  local filter=${2:-}
  local message="${3:-Existing central}"

  local existing_central=$(count-central "${filter}")
  if (( existing_central > 0 )); then
    echo
    echo "ERROR: Can't ${operation}.  ${message} (${existing_chassis} existing central)"
    exit 1
  fi
}

function check-no-chassis {
  local operation=$1
  local filter=${2:-}
  local message="${3:-Existing chassis}"

  local existing_chassis=$(count-chassis "${filter}")
  if (( existing_chassis > 0 )); then
    echo
    echo "ERROR: Can't ${operation}.  ${message} (${existing_chassis} existing chassis)"
    exit 1
  fi
}

function check-no-gw {
  local operation=$1
  local filter=${2:-}
  local message="${3:-Existing gw}"

  local existing_gw=$(count-gw "${filter}")
  if (( existing_gw > 0 )); then
    echo
    echo "ERROR: Can't ${operation}.  ${message} (${existing_gw} existing gw)"
    exit 1
  fi
}
....
function start() {
    echo "Starting OVN cluster"
    ovn_central=$1
    ovn_remote=$2
    ovn_add_chassis=$3

    if [ "x$ovn_central" == "x" ]; then
        ovn_central="yes"
    fi

    # Check that no ovn related containers are running if we're not adding
    # new containers.
    if [ "x$ovn_add_chassis" == "x" ]; then
        ovn_add_chassis="no"
    fi

    if [ "$ovn_central" == "yes" ]; then
        check-no-central "start"
        check-no-gw "start"
    fi

    if [ "$ovn_add_chassis" == "no" ] && [ "${CHASSIS_COUNT}" -gt 0 ]; then
        check-no-chassis "start"
    fi
    ...
    ```

Reply:
@dceara ack, I will fix it